grudge/reductions.py

```diff
@@ -156,6 +155,10 @@ def nodal_sum_loc(dcoll: DiscretizationCollection, dd, vec) -> float:
     :arg vec: a :class:`~meshmode.dof_array.DOFArray`.
     :returns: a scalar denoting the rank-local nodal sum.
     """
+    if isinstance(vec, np.ndarray):
```

Comment:
This should really use containers instead of special-casing `numpy` arrays. (below, too)